src/Common/ElapsedTimeProfileEventIncrement.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <Common/ProfileEvents.h>
+#include <Common/Stopwatch.h>
+
+namespace DB
+{
+
+enum Time
+{
+    Nanoseconds,
+    Microseconds,
+    Milliseconds,
+    Seconds,
+};
+
+template <Time time>
+struct ProfileEventTimeIncrement
+{
+    explicit ProfileEventTimeIncrement<time>(ProfileEvents::Event event_)
+        : event(event_), watch(CLOCK_MONOTONIC) {}
+
+    ~ProfileEventTimeIncrement()
+    {
+        watch.stop();
+        switch (time)
```

Comment:
I guess compiler can remove switch during compile time but it'd be more consistent to use `if constexpr` since `time` is template parameter